src/lib/storage/proposal/partition_creator.rb

```diff
@@ -217,7 +217,7 @@ def create_partition(vol, partition_id, free_slot)
             disk = ::Storage::Disk.find(devicegraph, free_slot.disk_name)
             ptable = disk.partition_table
             if logical_partition_preferred?(ptable)
-              create_extended_partition(disk, free_slot.region) unless ptable.has_extended
+              create_extended_partition(disk, free_slot.slot.region) unless ptable.has_extended
```

Comment:
NP: as can be seen from this bug it is quite confusing to have free_slot which is actually not slot and you have to call slot to get it.